File: frontend/src/types/customer.ts
/**
 * Customer Types and Interfaces
 */

export interface Customer {
  id: number;
  full_name: string;
  email: string;
  phone_number?: string;
  address?: string;
  created_at: string;
  updated_at: string;
}

export interface CustomerCreate {
  full_name: string;
  email: string;
  phone_number?: string;
  address?: string;
}

export interface CustomerUpdate {
  full_name?: string;
  phone_number?: string;
  address?: string;
}

export interface CustomerListResponse {
  total: number;
  page: number;
  page_size: number;
  items: Customer[];
}

File: backend/app/models/customer.py
"""
Customer model definition.
"""

from sqlalchemy import Column, Integer, String, Text, DateTime
from sqlalchemy.sql import func
from app.database import Base


class Customer(Base):
    """Customer model representing business customers."""
    
    __tablename__ = "customers"
    
    id = Column(Integer, primary_key=True, index=True)
    full_name = Column(String(255), nullable=False, index=True)
    email = Column(String(255), unique=True, nullable=False, index=True)
    phone_number = Column(String(20), nullable=True)
    address = Column(Text, nullable=True)
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now())
    
    def __repr__(self):
        return f"<Customer(id={self.id}, full_name={self.full_name}, email={self.email})>"
